ksp_plugin_adapter/ksp_plugin_adapter.cs

```diff
@@ -222,25 +222,15 @@ private enum UnityLayers {
       new Dictionary<uint, QP>();
 
   // UI for the apocalypse notification.
-  // The first apocalyptic error message.
-  [KSPField(isPersistant = true)]
-  private String revelation_ = "";
   // Whether we have encountered an apocalypse already.
   [KSPField(isPersistant = true)]
   private bool is_post_apocalyptic_ = false;
   [KSPField(isPersistant = true)]
-  private int apocalypse_dialog_x_ = Dialog.XCentre;
-  [KSPField(isPersistant = true)]
-  private int apocalypse_dialog_y_ = Dialog.YCentre;
   private Dialog apocalypse_dialog_ = new Dialog();
 
   // UI for the bad installation notification.
-  private String bad_installation_message_ = "";
-  private bool is_bad_installation_ = false;
+  private bool is_bad_installation_ = false;  // Don't persist.
   [KSPField(isPersistant = true)]
-  private int bad_installation_dialog_x_ = Dialog.XCentre;
-  [KSPField(isPersistant = true)]
-  private int bad_installation_dialog_y_ = Dialog.YCentre;
   private Dialog bad_installation_dialog_ = new Dialog();
```

Comment:
As discussed, the following are unsightly:
- a non-`null` invalid `Dialog`;
- a `bool` field equivalent to the `null`ness of another field.

Make the `Dialog`s `null` by default, remove the `is_post_apocalyptic_` field, create the `Dialog`s when the `bool`s would be set.
Take a message at construction in `Dialog`, so that there is no need for the `"SHOW WITHOUT MESSAGE"` thing.

For abstraction and readability, replace the `is_post_apocalyptic_` field with a property:
```C#
private bool is_post_apocalyptic { get { return apocalypse_dialog_ != null; } }
```
`is_bad_installation_` remains inequivalent to `bad_installation_dialog != null`; it can either become an auto-implemented property or remain a field.

```C#
if (is_post_apocalyptic_) {
  apocalypse_dialog_.Show();
}
```
becomes
```C#
apocalypse_dialog_?.Show();
```

Reply:
Done, except that `is_post_apocalyptic_` is really not needed anymore.